godot-core/src/docs.rs

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) godot-rust; Bromeon and contributors.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+macro_rules! define {
+    (struct $x:ident { $($f:ident),+ }) => {
+        #[derive(Clone, Copy, Debug, Default)]
+        pub struct $x {
+            $(pub $f: &'static str,)+
+        }
+    }
+}
+
+define!(struct DefinitionDocs { base, description, members });
+define!(struct ImplementationDocs { methods, signals, constants });
+#[cfg(since_api = "4.3")]
+#[derive(Default)]
+struct DocPieces {
+    definition: DefinitionDocs,
+    inherent: ImplementationDocs,
+    virtual_methods: &'static str,
+}
+
+#[cfg(since_api = "4.3")]
+#[doc(hidden)]
+pub fn gather() -> impl Iterator<Item = String> {
+    use crate::registry::plugin::PluginItem;
+    use std::collections::HashMap;
+    let mut map = HashMap::<&'static str, DocPieces>::new();
+    crate::private::iterate_plugins(|x| match x.item {
+        PluginItem::InherentImpl { docs, .. } => {
+            map.entry(x.class_name.as_str()).or_default().inherent = docs
+        }
+        PluginItem::ITraitImpl {
+            virtual_method_docs,
+            ..
+        } => {
+            map.entry(x.class_name.as_str())
+                .or_default()
+                .virtual_methods = virtual_method_docs
+        }
+        PluginItem::Struct { docs, .. } => {
+            map.entry(x.class_name.as_str()).or_default().definition = docs
+        }
+    });
+    map.into_iter().map(
+        |(
+            class,
+            DocPieces {
+                definition:
+                    DefinitionDocs {
+                        base,
+                        description,
+                        members,
+                    },
+                inherent:
+                    ImplementationDocs {
+                        methods,
+                        signals,
+                        constants,
+                    },
+                virtual_methods,
+            },
+        )| {
+            let brief = description.lines().next().unwrap_or_default();
```

Comment:
Please destructure those structs in the body, as individual variables. This parameter list is very hard to read.

Reply:
Like this?
```rust
            let DefinitionDocs {
                base,
                description,
                members,
            } = pieces.definition;
            let ImplementationDocs {
                methods,
                signals,
                constants,
            } = pieces.inherent;
            let virtual_methods = pieces.virtual_methods;
```

Reply:
Yes 🙂 

Please don't resolve if you're asking questions, it's easy to miss them otherwise.

Reply:
Well i applied that change, soo